service/gcs/transport/mock.go

```diff
@@ -46,3 +50,8 @@ func (c *MockConnection) CloseRead() error {
 func (c *MockConnection) CloseWrite() error {
 	return c.PipeWriter.Close()
 }
+
+// File returns a file that can be used to read/write to the connection.
+func (c *MockConnection) File() (*os.File, error) {
+	return nil, errors.New("not supported")
```

Comment:
Nitpicky, but you use the phrase "not implemented" in the error returned from mockReadWriteCloser's File method, but "not supported" here. It probably makes sense to use the same language for both.